tests/test_api_events_traversal_check.py

```
@@ -0,0 +1,53 @@
+    Test that stream_run_events detects path traversal attempts.
+    Before the fix, this test should fail (vulnerability confirmed).
+    After the fix, it should pass (validation error caught).
+    """
+
+    # Mock dependencies
+    with patch("swarm.api.server.get_spec_manager") as mock_get_manager, \
+         patch("swarm.runtime.resilient_db.check_db_health"):
+
+        # Setup mock manager
+        mock_manager = MagicMock()
+        mock_manager.runs_root = Path("/tmp/runs")
```

Ruff (0.14.14)

[error] Probable insecure usage of temporary file or directory: "/tmp/runs"

(S108)